test/stdlib/Net_HTTP_test.rb

```diff
@@ -1,50 +1,105 @@
+
+          while (line = s.gets) && !line.chomp.empty?
+          end
+
+          begin
+            s.write "HTTP/1.1 200 OK\r\n"
+            s.write "Content-Type: text/plain\r\n"
+            s.write "Content-Length: 0\r\n"
+            s.write "Connection: close\r\n"
+            s.write "\r\n"
+          rescue
+          ensure
+            s.close
+          end
+        end
+      end
+    end
+
+    def finish
+      @thread.kill
+      @server.close
+    end
+  end
+
+  def with_server(host)
+    server = Server.new(host)
+
+    res = nil
+    begin
+      res = yield server.uri
+    rescue
+    ensure
+      server.finish
+    end
+
+    res
+  end
+end
+
 class NetSingletonTest < Test::Unit::TestCase
   include TestHelper
+  include WithServer
 
   library "net-http", "uri"
   testing "singleton(::Net::HTTP)"
 
   def test_get
     $stdout = StringIO.new
-    assert_send_type "(URI::Generic) -> nil",
-                     Net::HTTP, :get_print, URI("https://www.ruby-lang.org")
-    assert_send_type "(String, String) -> nil",
-                     Net::HTTP, :get_print, 'www.ruby-lang.org', '/en'
-    assert_send_type "(URI::Generic, Hash[String, String]) -> nil",
-                     Net::HTTP, :get_print, URI("https://www.ruby-lang.org"), {"Accept" => "text/html"}
-    assert_send_type "(URI::Generic, Hash[Symbol, String]) -> nil",
-                     Net::HTTP, :get_print, URI("https://www.ruby-lang.org"), {Accept: "text/html"}
-    assert_send_type "(URI::Generic) -> String",
-                     Net::HTTP, :get, URI("https://www.ruby-lang.org")
-    assert_send_type "(String, String) -> String",
-                     Net::HTTP, :get, 'www.ruby-lang.org', '/en'
-    assert_send_type "(URI::Generic, Hash[String, String]) -> String",
-                     Net::HTTP, :get, URI("https://www.ruby-lang.org"), {"Accept" => "text/html"}
-    assert_send_type "(URI::Generic, Hash[Symbol, String]) -> String",
-                     Net::HTTP, :get, URI("https://www.ruby-lang.org"), {Accept: "text/html"}
-    assert_send_type "(URI::Generic) -> Net::HTTPResponse",
-                     Net::HTTP, :get_response, URI("https://www.ruby-lang.org")
-    assert_send_type "(String, String) -> Net::HTTPResponse",
-                     Net::HTTP, :get_response, 'www.ruby-lang.org', '/en'
-    assert_send_type "(URI::Generic, Hash[String, String]) -> Net::HTTPResponse",
-                     Net::HTTP, :get_response, URI("https://www.ruby-lang.org"), {"Accept" => "text/html"}
-    assert_send_type "(URI::Generic, Hash[Symbol, String]) -> Net::HTTPResponse",
-                     Net::HTTP, :get_response, URI("https://www.ruby-lang.org"), {Accept: "text/html"}
+    with_server("localhost") do |uri|
+      assert_send_type "(URI::Generic) -> nil",
+                       Net::HTTP, :get_print, uri
+      assert_send_type "(String, String) -> nil",
+                       Net::HTTP, :get_print, "www.ruby-lang.org", "/en"
```

Comment:
Some lines look like they still connect to `ruby-lang.org`. Is this intentional?

Reply:
I was testing while disconnecting the PC from the network, but it seems the error was being suppressed in the `rescue` block.